frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/print/PrintElementGeneratorImpl.java

```diff
@@ -0,0 +1,106 @@
+package com.virtuslab.gitmachete.frontend.graph.print;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import com.intellij.util.SmartList;
+import com.intellij.vcs.log.graph.EdgePrintElement.Type;
+import com.intellij.vcs.log.graph.api.EdgeFilter;
+import com.intellij.vcs.log.graph.api.elements.GraphEdge;
+import com.intellij.vcs.log.graph.api.elements.GraphEdgeType;
+import com.intellij.vcs.log.graph.api.elements.GraphNode;
+import com.intellij.vcs.log.graph.api.printer.PrintElementGenerator;
+import com.intellij.vcs.log.graph.impl.print.elements.EdgePrintElementImpl;
+import com.intellij.vcs.log.graph.impl.print.elements.PrintElementWithGraphElement;
+import com.intellij.vcs.log.graph.impl.print.elements.SimplePrintElementImpl;
+import lombok.RequiredArgsConstructor;
+import org.checkerframework.checker.index.qual.NonNegative;
+
+import com.virtuslab.gitmachete.frontend.graph.GraphElementManager;
+import com.virtuslab.gitmachete.frontend.graph.nodes.BranchNode;
+import com.virtuslab.gitmachete.frontend.graph.nodes.IGraphNode;
+import com.virtuslab.gitmachete.frontend.graph.repository.RepositoryGraph;
+
+@RequiredArgsConstructor
+public final class PrintElementGeneratorImpl implements PrintElementGenerator {
+  private final RepositoryGraph repositoryGraph;
+  private final GraphElementManager printElementManager;
+
+  @Override
+  public Collection<PrintElementWithGraphElement> getPrintElements(int rowIndex) {
+    assert rowIndex >= 0;
+    PrintElementBuilder builder = new PrintElementBuilder(rowIndex);
+    collectElements(rowIndex, builder);
+    return builder.build();
+  }
+
+  private void collectElements(@NonNegative int rowIndex, PrintElementBuilder builder) {
+    GraphNode graphNode = new GraphNode(rowIndex);
+    IGraphNode graphElement = repositoryGraph.getGraphNode(rowIndex);
```

Comment:
Hmm `graphNode` can as well be called `graphElement` (coz `GraphNode implements GraphElement`)... and in fact `graphElement` should not be called this way (b/c `com.intellij.vcs.log.graph.api.elements.GraphElement` belongs to an unrelated hierarchy AFAICS... pls refine the naming; also, the only use of `graphNode` is at the far end of this method

Reply:
Also, this poses the question of what's the relation b/w `com.intellij.vcs.log.graph.api.elements.GraphNode` and `com.virtuslab.gitmachete.frontend.graph.nodes.{I, Base}GraphNode`... and whether the latter should be called in a different way (?)

Reply:
I renamed variables and moved `GraphNode` declaration closer to its use to minimize the confusion. It sadly seems that it (the confusion) cannot be fully eliminated since we need to use ` com.intellij.vcs.log.graph.api.elements.GraphNode` here.

Reply:
`GraphNode` (and `GraphEdge`) is like an adapter between our "logical" `IGraphNode`s and `PrintElement`s, objects containing data about what and where draw ("print"). 

So the above question is actually about where is the border between us and them (intellij). For now, I can assume that we could get rid of `GraphNode` (and `GraphEdge`) and implement our own logic around `com.intellij.vcs.log.graph.PrintElement`.

Reply:
I think that the further (deeper) "expansion" of our code is not a part of this issue(s). This PR is huge anyway.  I would rather suppose to try it again by the way of another issue or create a dedicated issue. WDYT?

Reply:
https://github.com/VirtusLab/git-machete-intellij-plugin/issues/212